src/libs/OptionsListUtils.ts

```diff
@@ -355,79 +453,77 @@ function getSearchText(report, reportName, personalDetailList, isChatRoomOrPolic
     let lastMessageTextFromReport = '';
-    const lastActionName = lodashGet(lastReportAction, 'actionName', '');
+    const lastActionName = lastReportAction?.actionName ?? '';
 
     if (ReportActionUtils.isMoneyRequestAction(lastReportAction)) {
         const properSchemaForMoneyRequestMessage = ReportUtils.getReportPreviewMessage(report, lastReportAction, true, false, null, true);
         lastMessageTextFromReport = ReportUtils.formatReportLastMessageText(properSchemaForMoneyRequestMessage);
     } else if (ReportActionUtils.isReportPreviewAction(lastReportAction)) {
         const iouReport = ReportUtils.getReport(ReportActionUtils.getIOUReportIDFromReportActionPreview(lastReportAction));
-        const lastIOUMoneyReportAction = _.find(
-            allSortedReportActions[iouReport.reportID],
+        const lastIOUMoneyReportAction = allSortedReportActions[iouReport?.reportID ?? '']?.find(
             (reportAction, key) =>
                 ReportActionUtils.shouldReportActionBeVisible(reportAction, key) &&
                 reportAction.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE &&
                 ReportActionUtils.isMoneyRequestAction(reportAction),
         );
-        lastMessageTextFromReport = ReportUtils.getReportPreviewMessage(iouReport, lastIOUMoneyReportAction, true, ReportUtils.isChatReport(report), null, true);
+        lastMessageTextFromReport = ReportUtils.getReportPreviewMessage(
+            !isEmptyObject(iouReport) ? iouReport : null,
+            lastIOUMoneyReportAction,
+            true,
+            ReportUtils.isChatReport(report),
+            null,
+            true,
+        );
     } else if (ReportActionUtils.isReimbursementQueuedAction(lastReportAction)) {
         lastMessageTextFromReport = ReportUtils.getReimbursementQueuedActionMessage(lastReportAction, report);
     } else if (ReportActionUtils.isReimbursementDeQueuedAction(lastReportAction)) {
         lastMessageTextFromReport = ReportUtils.getReimbursementDeQueuedActionMessage(report);
     } else if (ReportActionUtils.isDeletedParentAction(lastReportAction) && ReportUtils.isChatReport(report)) {
         lastMessageTextFromReport = ReportUtils.getDeletedParentActionMessageForChatReport(lastReportAction);
-    } else if (ReportUtils.isReportMessageAttachment({text: report.lastMessageText, html: report.lastMessageHtml, translationKey: report.lastMessageTranslationKey})) {
-        lastMessageTextFromReport = `[${Localize.translateLocal(report.lastMessageTranslationKey || 'common.attachment')}]`;
+    } else if (ReportUtils.isReportMessageAttachment({text: report?.lastMessageText ?? '', html: report?.lastMessageHtml, translationKey: report?.lastMessageTranslationKey, type: ''})) {
+        lastMessageTextFromReport = `[${Localize.translateLocal((report?.lastMessageTranslationKey ?? 'common.attachment') as TranslationPaths)}]`;
```

Comment:
This caused regression. We should keep `||`

BUG: Crash on LHN for non-expensify email domain, `[MISSING TRNSLATION]` for expensify domain
`[Attachment]` is the expected string

<img width="330" alt="Screenshot 2024-01-28 at 10 58 29 PM" src="https://github.com/Expensify/App/assets/108292595/c4fd5f02-fb06-4738-bdc0-fa6b1ccf64a7">

<img width="483" alt="Screenshot 2024-01-28 at 10 58 10 PM" src="https://github.com/Expensify/App/assets/108292595/ac261b0e-925b-4d8a-8c94-48f8de9714a0">

<img width="323" alt="Screenshot 2024-01-28 at 11 01 08 PM" src="https://github.com/Expensify/App/assets/108292595/5fd76157-50ed-46af-9a77-4089594a96ab">

Repro step: send any attachment to any chat on iOS app